libraries/chain/include/eosio/chain/fork_database.hpp

```diff
@@ -171,6 +163,27 @@ namespace eosio::chain {
       // see fork_database_t::fetch_branch(forkdb->head()->id())
       block_branch_t fetch_branch_from_head() const;
 
+      block_id_type pending_lib_id(const block_id_type& head_id) const {
+         if (in_use.load() == in_use_t::legacy) {
+            block_state_legacy_ptr head;
+            if (head_id.empty()) {
+               head = fork_db_l.head();
+            } else {
+               // maintain legacy only advancing LIB via validated blocks
```

Comment:
I'm not quite understanding how this comment is accurate, since at line 170 above we use `fork_db_l.head();` (which I think doesn't require the blocks to be validated anymore)?

Reply:
But if the function is called with an empty id like line 170, then we use `fork_db_l.head()` which may not be valid.